build.gradle

```diff
@@ -53,35 +41,27 @@ dependencies {
     testDep 'org.hamcrest:hamcrest:2.1'
     testDep 'org.jacoco:org.jacoco.core:0.8.3'
     testDep 'org.jacoco:org.jacoco.report:0.8.3'
-
-    awslc "awslc-src:FIPS:${awslc_version}@tar.gz"
 }
 
 defaultTasks 'release'
 
-task getAwsLcSrc(type: Copy) {
-    def tarPath = project.configurations.awslc.find {it.name.startsWith("FIPS") }
-    def tarFile = file(tarPath)
-
-    doFirst {
-        exec {
-            workingDir tarFile.parent
-            commandLine 'sha256sum', '--check', "${projectDir}/awslc.sha256"
-        }
-
-        mkdir "${buildDir}/awslc/bin"
+// NOTE: implemented as a function because gradle didn't appear to block on git
+//       commands until completion when run as a task
```

Comment:
Shouldn't ensuring the sub-module is updated be the responsibility of the developer running git rather than gradle?

Reply:
sort of. this command actually pulls the submodule code into the ACCP source tree. without running this command, `./aws-lc` is just an empty directory and ACCP won't build. so, without this logic, ACCP users would need to either clone the ACCP repo with `git clone --recursive` (which is non-standard), or they'd need to run the `git submodule` command(s) themselves. my goal here was to keep the same workflow as the curl/tar approach -- just run `./gradlew release` after cloning the repo and you're all set.

i suppose one disadvantage of the implementation as-written is that it introduces a dependency on `git` (which is why we're seeing the codebuild failures -- that build environment doesn't have git installed), but we already have external dependencies not modeled in `gradle.build` et al. (cmake, gradle, etc.) so this doesn't seem like a significant imposition.

of course, this submodule fetching logic is trivial to remove. let me know what you think.

Reply:
I think that we can just document that clones need to be recursive. It might make sense to check for an empty `aws-lc` directory in `build.gradle` and return a useful error message though.

Reply:
alright. that should be easy enough to do. i'll post another revision.